src/Symfony/Component/Messenger/Handler/MessageHandlerInterface.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Handler;
+
+/**
+ * Handlers can implements this interface. This allows them to be auto-configured and to
+ * handle multiple messages.
+ *
+ * @author Samuel Roze <samuel.roze@gmail.com>
+ */
+interface MessageHandlerInterface
+{
+    /**
+     * Return a list of messages to be handled.
+     *
+     * It returns a list of messages like in the following example:
+     *
+     *     return [MyMessage::class];
+     *
+     * It can also change the priority per classes.
+     *
+     *     return [
+     *         [FirstMessage::class, 0],
+     *         [SecondMessage::class, -10],
```

Comment:
I do not like priorities. But we've discussed that earlier...

However, what I do miss is the fact that a single handler could handle two different messages in two different functions. 

```
return [
  FirstMessage::class => 'myFunction',
  SecondMessage::class => ['myOtherFunction', -10],
];

Reply:
I thought about this potential use case of using different functions. But I don’t see a real value of adding this now to be honest. I believe that if a handler gets multiple messages they would be very similar and `instanceof`s within the `__invoke` method should be enough.

However, if we are many to believe this should be added I’m happy to update the PR.